### PR TITLE
✨ account_id and name on sequences

### PR DESCRIPTION
**Pull request by RTLS**

### CODE REVIEW

**Comment by RTLS** [comment]
* **#516** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/516?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @RTLS and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: priv/repo/migrations/20241107205658_add_account_id_and_name_to_sequences.exs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      modify :account_id, :uuid, null: false
+      modify :name, :string, null: false
+
+      modify :postgres_database_id,
+             references(:postgres_databases,
+               with: [account_id: :account_id],
+               prefix: @config_schema
+             ),
+             null: false
+    end
```
Comment (RTLS) [inline priv/repo/migrations/20241107205658_add_account_id_and_name_to_sequences.exs:47]: I drop the constraint to then add it back as a composite foreign key to enforce that account_id matches on databases and sequences

File: priv/repo/migrations/20241107205658_add_account_id_and_name_to_sequences.exs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    # Set name using composed format: database_name.schema.table_name
+    execute("""
+    UPDATE #{@config_schema}.sequences s
+    SET name = (
+      SELECT pd.name || '.' || s2.table_schema || '.' || s2.table_name
+      FROM #{@config_schema}.sequences s2
+      JOIN #{@config_schema}.postgres_databases pd ON s2.postgres_database_id = pd.id
+      WHERE s2.id = s.id
+    )
+    """)
```
Comment (RTLS) [inline priv/repo/migrations/20241107205658_add_account_id_and_name_to_sequences.exs:33]: ran this on prod db and the generated names are:

```
supabase.public.test
escape-social.public.notifications
coding-interview.public.profile
quiet_pine_43.public.users
rough-wood-45.public.countries
ancient-wave-45.public.customer
idojobase-sequin.public.group_member_contentom_answers
neon.public.users
callback.public.doku_notification_non_snap
superai.storage.objects
supabase-experimental.cron.job_run_details
supabase-experimental.public.reservations
supabase-experimental.public.vehicles
supabase-experimental.public.relocations
ep-yellow-sun-a20mq229.public.article_metadata
snowy_bush.public.addresses
devapi.public.sequin_events
prodapi.public.sequin_events
polished_waterfall_51.public.foo
ep-yellow-sun-a20mq229.public.sequin_events
supabase-experimental.public.sequin_events
superai.public.sequin_events
callback.public.sequin_events
neon_db.public.orders
icy_shadow_29.demo.orders
icy_shadow_29.demo.sequin_events
timescale_local2.public.sensor_readings
dune-local.public.characters
wandering_cloud_67.public.items
morning_fire_24.public.messages
morning_fire_24.public.leads
```

Comment (acco) [reply priv/repo/migrations/20241107205658_add_account_id_and_name_to_sequences.exs:33]: good call!

File: priv/repo/migrations/20241107205658_add_account_id_and_name_to_sequences.exs [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+      modify :postgres_database_id,
+             references(:postgres_databases,
+               with: [account_id: :account_id],
+               prefix: @config_schema
+             ),
+             null: false
+    end
+
+    create unique_index(:sequences, [:account_id, :id], prefix: @config_schema)
```
Comment (acco) [inline priv/repo/migrations/20241107205658_add_account_id_and_name_to_sequences.exs:49]: what's this one about? is this for the with constraint?

**Review by acco — APPROVED** [review]

File: priv/repo/migrations/20241107205658_add_account_id_and_name_to_sequences.exs (hunk 3 above)
Comment (RTLS) [reply priv/repo/migrations/20241107205658_add_account_id_and_name_to_sequences.exs:49]: need it for the composite foreign key back from consumers

**Pull request merged**